prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt

```diff
@@ -135,22 +135,7 @@ class FHIRRouter(
 
         var finalReceivers = mutableListOf<Receiver>()
         if (listOfReceivers.isNotEmpty()) {
-            if (message.topic == Topic.SEND_ORIGINAL) {
-                listOfReceivers.forEach { receiver ->
-                    val originalMessageFormat = getOriginalMessageBodyFormat(message.reportId, WorkflowEngine())
-                    if ((originalMessageFormat == "HL7" && receiver.format == Report.Format.HL7) ||
-                        (originalMessageFormat == "FHIR" && receiver.format == Report.Format.FHIR)
-                    ) {
-                        finalReceivers.add(receiver)
-                    }
-                }
-            } else {
-                finalReceivers = listOfReceivers.toMutableList()
-            }
-
-            if (finalReceivers.isEmpty()) {
-                ActionHistory(TaskAction.process_warning, true)
-            }
+            finalReceivers = findReceiversForBundle(message, listOfReceivers, finalReceivers)
```

Comment:
Sorry for the confusion, but I was actually suggesting combining this logic into the `findReceiversForBundle` function called at line 134. I think we ended up in the current branch having two `findReceiversForBundle` functions, so basically just combining those into one, so that here in `doWork` we only need to get one list of receivers, we don't need `listOfReceivers` and `finalReceivers`.

Reply:
Oh, the reason these are separate is because Arnej asked for it to be separate because he wanted there to be a separate ActionHistory for it and for it not to be seen as part of the filters. Unless I misunderstood his and my conversation @arnejduranovic ?